src/gui/dialogs/statistics_dialog.cpp

```diff
@@ -164,19 +183,83 @@ void statistics_dialog::add_damage_row(
 // Return the string to use in the "Hits" table, showing actual and expected number of hits.
 static std::string tally(const std::map<int, struct statistics::stats::by_cth_t>& by_cth)
 {
-	int overall_hits = 0;
+	unsigned int overall_hits = 0;
 	double expected_hits = 0;
+	unsigned int overall_strikes = 0;
 
 	for(const auto& i : by_cth) {
 		int cth = i.first;
 		overall_hits += i.second.hits;
 		expected_hits += (cth * 0.01) * i.second.strikes;
+		overall_strikes += i.second.strikes;
 	}
 
-	// TODO: show a priori probability of this actual result here. https://en.wikipedia.org/wiki/Poisson_binomial_distribution
-
 	std::ostringstream str;
 	str << overall_hits << " / " << expected_hits;
+
+	// Compute the a priori probability of this actual result, by simulating many attacks against a single defender.
+	{
+		config defender_cfg(
+			"id", "statistics_dialog_dummy_defender",
+			"hide_help", "yes",
+			"do_not_list", "yes",
+			"hitpoints", overall_strikes
+		);
+		unit_type defender_type(defender_cfg);
+		unit_types.build_unit_type(defender_type, unit_type::BUILD_STATUS::FULL);
+
+		battle_context_unit_stats defender_bc(&defender_type, nullptr, false, nullptr, nullptr, 0 /* not used */);
+		std::unique_ptr<combatant> current_defender(new combatant(defender_bc));
+
+		for(const auto& i : by_cth) {
+			int cth = i.first;
+			config attacker_cfg(
+				"id", "statistics_dialog_dummy_attacker" + std::to_string(cth),
+				"hide_help", "yes",
+				"do_not_list", "yes",
```

Comment:
Done.  I also found another instance but did not change it since I was not 100% sure the change would be correct there: https://github.com/wesnoth/wesnoth/blob/40fe145213b42a56df68e2c63f16714751bdc39f/src/statistics.cpp#L621

Reply:
The change would indeed be correct there as well.

Reply:
Done: 39bdda9d256